src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/CodeStyle/CSharpIdeCodeStyleOptions.cs

```diff
@@ -80,6 +79,7 @@ internal sealed record class CSharpIdeCodeStyleOptions : IdeCodeStyleOptions, IE
     [DataMember] public CodeStyleOption2<bool> PreferReadOnlyStruct { get; init; } = CodeStyleOption2.TrueWithSuggestionEnforcement;
     [DataMember] public CodeStyleOption2<bool> PreferReadOnlyStructMember { get; init; } = CodeStyleOption2.TrueWithSuggestionEnforcement;
     [DataMember] public CodeStyleOption2<bool> PreferStaticLocalFunction { get; init; } = CodeStyleOption2.TrueWithSuggestionEnforcement;
+    [DataMember] public CodeStyleOption2<bool> PreferStaticAnonymousFunction { get; init; } = CodeStyleOption2.TrueWithSuggestionEnforcement;
```

Comment:
This needs to be off by default. It is way too noisy. People need to opt into this.

Reply:
`AbstractBuiltInCodeStyleDiagnosticAnalyzer` by default makes reported diagnostics `Hidden`. Thus _the only_ way to observe its presence is to invoke a code fix menu and see a proposed fix there:
![devenv_Wkfj6ewcil](https://github.com/dotnet/roslyn/assets/70431552/a5e71e94-5cab-4b46-8cf8-01a5695b3f1f)
I don't think this can be considered "noisy". Do you?

Reply:
That's ok then.  The fixer should lower the priority though in that case. It should be normal if visible, and low if not. Thanks!

Reply:
> It should be normal if visible, and low if not

Does this apply only to this fixer or we want such logic to apply to all fixers? Because if the latter than it's probably worth its own PR dedicated specificly to that

Reply:
it applies to this one.

Reply:
> AbstractBuiltInCodeStyleDiagnosticAnalyzer by default makes reported diagnostics Hidden

can you show me where that happens?

Reply:
Yeah, here:
https://github.com/dotnet/roslyn/blob/d20c3e1ecbc2e239ef394bb1be5ea57701368694/src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer_Core.cs#L52-L70

Reply:
Done (though in this particular case I would probably prefer new fixer over "Introduce local"):
![devenv_S7cBNuSALU](https://github.com/dotnet/roslyn/assets/70431552/dc442040-eb31-40bc-9717-44f90b4eb077)